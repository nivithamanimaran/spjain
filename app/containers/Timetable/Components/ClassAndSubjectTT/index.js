import React, { useContext, useEffect, useState } from 'react';
import TTCell from '../TTCell';
import TTGrid from '../TTGrid';
import { generateTimeSlots } from '../../../../utils/helpers/ttHelper';
import { Box, Typography } from '@material-ui/core';
import {
  getAllTimeBlocks,
  getTimetableByContactId,
} from '../../../../utils/ApiService';
import { makeStyles } from '@material-ui/core/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import WeekPicker from '../WeekPicker';
import KenLoader from '../../../../components/KenLoader';
import { useTranslation } from 'react-i18next';
import configContext from '../../../../utils/helpers/configHelper';
import KenTabs from '../../../../components/KenTabs';
import {
  getDayStartEnd,
  formatData,
  changeWeek,
  isDateNew,
  getCurrentWeek,
} from '../../helper';

const useStyles = makeStyles(theme => ({
  selectorStyle: {
    display: 'table-cell',
    padding: '0px 16px',
    fontSize: '12px',
    textAlign: 'left',
    fontFamily: 'Work Sans,sans-serif',
    fontWeight: 400,
    lineHeight: 1,
    verticalAlign: 'inherit',
  },
  timeBox: {
    border: '1px solid lightgray',
    borderRadius: '3px',
  },
  timeText: {
    fontSize: '13px',
    padding: '5px',
    minWidth: '120px',
  },
  classSectionCell: {
    minWidth: 'max-content',
    fontSize: '13px',
  },
  classCell: {
    padding: '22px',
    fontSize: '14px',
    minWidth: 'max-content',
    border: '1px solid lightGrey',
    borderRadius: '3px',
    textAlign: 'center',
  },
  arrow: {
    margin: '0px 10px',
    width: '16px',
    cursor: 'pointer',
  },
  dayHeader: {
    fontSize: '16px',
    textAlign: 'start',
    paddingLeft: '10px',
    fontWeight: 500,
  },
  dateSelector: {
    fontSize: '14px',
  },
  tableContainer: {
    display: 'flex',
    justifyContent: 'flex-start',
  },
  tableWrap: {
    overflow: 'auto',
    '&::-webkit-scrollbar': {
      height: '4px',
    },
    '&::-webkit-scrollbar-track': {
      boxShadow: 'inset 0 0 6px rgba(0,0,0,0.00)',
      backgroundColor: 'rgba(0,0,0,.2)',
    },
    '&::-webkit-scrollbar-thumb': {
      backgroundColor: '#252628b0',
    },
  },
  ttPlaceholder: {
    position: 'absolute',
    top: '50%',
    left: '50%',
  },
  ttLabel: {
    color: theme.palette.KenColors.neutral900,
    fontSize: 18,
    fontWeight: 600,
  },
}));

function ClassAndSubjectTT(props) {
  const { user = { AccountId: '', ContactId: '' } } = props;
  const { config } = useContext(configContext);
  const { t } = useTranslation();
  const classes = useStyles();
  const [currentWeek, setCurrentWeek] = useState([]);
  const [timeBlocksDuration] = useState(config?.timetable?.timeBlocksDuration);
  const [timings, setTimings] = useState();
  const [dateToViewTT, setDateToViewTT] = useState(new Date());
  const [loading, setLoading] = useState(false);
  const [tabData, setTabData] = React.useState();
  const [classTTData, setClassTTData] = React.useState();
  const [subjectsTTData, setSubjectsTTData] = React.useState();

  const handleDateChange = date => {
    setDateToViewTT(new Date(date));
  };

  //Get All time blocks
  useEffect(() => {
    const { AccountId } = user;
    //get time blocks
    if (AccountId) {
      getAllTimeBlocks(AccountId)
        .then(res => {
          const dayTimings = getDayStartEnd(res);
          setTimings(dayTimings);
        })
        .catch(err => {
          console.log('error in get time blocks::', err);
        });
    }
  }, [user]);

  useEffect(() => {
    setLoading(true);
    //get user details
    const { AccountId, ContactId } = user;

    //Recall API only if date dose not exists in current week

    if (isDateNew(currentWeek, dateToViewTT)) {
      //get timetable
      if (dateToViewTT && AccountId && ContactId) {
        // getTimetableByContactId(dateToViewTT, AccountId, ContactId)
        getTimetableByContactId(dateToViewTT, AccountId, '0031y00000DnEZBAA3') //Aradhya Pandey
          .then(res => {
            setLoading(false);
            const formattedData = formatData(res, dateToViewTT);
            setClassTTData(formattedData);
          })
          .catch(err => {
            console.log('error in getTimetableByContactId', err);
            setLoading(false);
            setClassTTData([]);
          });

        getTimetableByContactId(dateToViewTT, AccountId, '0031y00000DnMfvAAF') //Masthan Faculty
          .then(res => {
            setLoading(false);
            const formattedData = formatData(res, dateToViewTT);
            setSubjectsTTData(formattedData);
          })
          .catch(err => {
            console.log('error in getTimetableByContactId', err);
            setLoading(false);
            setSubjectsTTData([]);
          });
      } else {
        setLoading(false);
      }
    } else {
      setLoading(false);
    }
  }, [dateToViewTT]);

  useEffect(() => {
    const week = getCurrentWeek(dateToViewTT);
    setCurrentWeek(week);
  }, [dateToViewTT]);

  useEffect(() => {
    const tabs = [
      {
        title: t('headings:Subject_TT'),
        content: (
          <TTGrid columnsArr={subjectsColumns} dataArr={subjectsTTData} />
        ),
      },
      {
        title: t('headings:Class_TT'),
        content: <TTGrid columnsArr={classColumns} dataArr={classTTData} />,
      },
    ];
    setTabData(tabs);
  }, [subjectsTTData, classTTData]);

  const classColumns = React.useMemo(
    () => [
      {
        Header: '',
        id: 'day',
        accessor: 'day',
        Cell: ({ row, value }) => {
          return (
            <td className={classes.selectorStyle}>
              <Typography className={classes.classCell}>
                {value?.toUpperCase()?.slice(0, 3)}
              </Typography>
            </td>
          );
        },
      },
      ...generateTimeSlots(
        timings?.StartTime,
        timings?.EndTime,
        timeBlocksDuration
      ).map(s => ({
        Header: (
          <Typography className={classes.timeBox}>
            <Typography className={classes.timeText}>{s.header}</Typography>
          </Typography>
        ),
        Cell: cellProps => {
          const { row } = cellProps;
          return TTCell(
            s.startTime,
            s.endTime,
            row?.original?.date,
            'student',
            timeBlocksDuration
          )(cellProps);
        },
        id: s.header,
      })),
    ],
    [timings]
  );
  const subjectsColumns = React.useMemo(
    () => [
      {
        Header: '',
        id: 'day',
        accessor: 'day',
        Cell: ({ row, value }) => (
          <td className={classes.selectorStyle}>
            <Typography className={classes.classCell}>
              {value?.toUpperCase()?.slice(0, 3)}
            </Typography>
          </td>
        ),
      },
      ...generateTimeSlots(
        timings?.StartTime,
        timings?.EndTime,
        timeBlocksDuration
      ).map(s => ({
        Header: (
          <Typography className={classes.timeBox}>
            <Typography className={classes.timeText}>{s.header}</Typography>
          </Typography>
        ),
        Cell: cellProps => {
          const { row } = cellProps;
          return TTCell(
            s.startTime,
            s.endTime,
            row?.original?.date,
            'faculty',
            timeBlocksDuration
          )(cellProps);
        },
        id: s.header,
      })),
    ],
    [timings]
  );

  return (
    <Box>
      {loading && <KenLoader />}
      <Box>
        <Typography className={classes.ttLabel}>
          {t('headings:Subject_Class_TT')}
        </Typography>
      </Box>
      {/* <TTHeader /> */}
      <Box
        style={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'flex-end',
        }}
      >
        <Box
          style={{
            display: 'flex',
            alignItems: 'baseline',
          }}
        >
          <Box className={classes.dateSelector}>
            <ArrowBackIcon
              className={classes.arrow}
              size="small"
              onClick={() => changeWeek('prev')}
            />
          </Box>
          <WeekPicker
            value={dateToViewTT}
            onChange={handleDateChange}
            selector={'week'}
            inputSize="10"
          />
          <Box className={classes.dateSelector}>
            <ArrowForwardIcon
              className={classes.arrow}
              size="small"
              onClick={() => changeWeek('next')}
            />
          </Box>
        </Box>
      </Box>
      <Box p={2}>
        <>
          <Box className={classes.tableContainer}>
            <Box className={classes.tableWrap}>
              {tabData && (
                <KenTabs
                  data={tabData}
                  tabPanelProps={{ style: { overflow: 'auto' } }}
                />
              )}
            </Box>
          </Box>
        </>
      </Box>
    </Box>
  );
}

export default ClassAndSubjectTT;
