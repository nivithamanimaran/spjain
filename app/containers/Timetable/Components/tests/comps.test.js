import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import ThemeProvider from '../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

import CommonTT from '../CommonTT/index';
import ClassAndSubjectTT from '../ClassAndSubjectTT/index';
import TTHeader from '../TTHeader';
import WeekPicker from '../WeekPicker';
import TTGrid from '../TTGrid';

describe('<CommonTT />', () => {
  it('CommonTT should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <Router>
            <CommonTT />
          </Router>
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<ClassAndSubjectTT />', () => {
  it('ClassAndSubjectTT should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <Router>
            <ClassAndSubjectTT />
          </Router>
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<TTHeader />', () => {
  it('TTHeader should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <Router>
            <TTHeader />
          </Router>
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<WeekPicker />', () => {
  it('WeekPicker should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <Router>
            <WeekPicker />
          </Router>
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<TTGrid />', () => {
  it('TTGrid should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <Router>
            <TTGrid />
          </Router>
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
