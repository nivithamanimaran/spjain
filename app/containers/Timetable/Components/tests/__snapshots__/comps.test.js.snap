// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassAndSubjectTT /> ClassAndSubjectTT should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-110"
>
  <div
    class="MuiBox-root MuiBox-root-111"
  >
    <p
      class="MuiTypography-root makeStyles-ttLabel-109 MuiTypography-body1"
    >
      headings:Subject_Class_TT
    </p>
  </div>
  <div
    class="MuiBox-root MuiBox-root-112"
    style="display: flex; align-items: center; justify-content: flex-end;"
  >
    <div
      class="MuiBox-root MuiBox-root-113"
      style="display: flex; align-items: baseline;"
    >
      <div
        class="MuiBox-root MuiBox-root-114 makeStyles-dateSelector-105"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-arrow-103"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
      <div
        class="MuiFormControl-root-123 MuiTextField-root-122"
        format="MMMM do"
        inputsize="10"
        selector="week"
      >
        <label
          class="MuiFormLabel-root-135 MuiInputLabel-root-127 MuiInputLabel-formControl-129 MuiInputLabel-animated-132 MuiInputLabel-shrink-131 MuiFormLabel-filled-137"
          data-shrink="true"
        />
        <div
          class="MuiInputBase-root-150 MuiInput-root-139 MuiInput-underline-142 MuiInputBase-formControl-151 MuiInput-formControl-140"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-158 MuiInput-input-146"
            readonly=""
            size="10"
            type="text"
            value="AUGUST-08-2021 TO AUGUST-14-2021"
          />
        </div>
        
      </div>
      <div
        class="MuiBox-root MuiBox-root-184 makeStyles-dateSelector-105"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-arrow-103"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-185"
  >
    <div
      class="MuiBox-root MuiBox-root-186 makeStyles-tableContainer-106"
    >
      <div
        class="MuiBox-root MuiBox-root-187 makeStyles-tableWrap-107"
      >
        <div
          class="makeStyles-root-188"
        >
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault makeStyles-appBar-190 MuiPaper-elevation0"
          >
            <div
              class="MuiTabs-root makeStyles-tabs-189"
            >
              <div
                class="MuiTabs-scrollable"
                style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
              />
              <div
                class="MuiTabs-scroller MuiTabs-scrollable"
                style="margin-bottom: 0px;"
              >
                <div
                  aria-label="scrollable auto tabs example"
                  class="MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="scrollable-auto-tabpanel-0"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                    id="scrollable-auto-tab-0"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper"
                    >
                      headings:Subject_TT
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="scrollable-auto-tabpanel-1"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
                    id="scrollable-auto-tab-1"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper"
                    >
                      headings:Class_TT
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="PrivateTabIndicator-root-191 PrivateTabIndicator-colorPrimary-192 MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </header>
          <div
            aria-labelledby="scrollable-auto-tab-0"
            id="scrollable-auto-tabpanel-0"
            role="tabpanel"
            style="overflow: auto;"
          >
            <div
              class="MuiBox-root MuiBox-root-195"
            >
              <table
                class="makeStyles-table-199"
                role="table"
              >
                <thead>
                  <tr
                    class="makeStyles-tr-197"
                    role="row"
                  >
                    <th
                      colspan="1"
                      role="columnheader"
                    />
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          09:00 09:15
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          09:15 09:30
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          09:30 09:45
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          09:45 10:00
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          10:00 10:15
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          10:15 10:30
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          10:30 10:45
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          10:45 11:00
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          11:00 11:15
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          11:15 11:30
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          11:30 11:45
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          11:45 12:00
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          12:00 12:15
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          12:15 12:30
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          12:30 12:45
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          12:45 13:00
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          13:00 13:15
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          13:15 13:30
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          13:30 13:45
                        </p>
                      </p>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      <p
                        class="MuiTypography-root makeStyles-timeBox-99 MuiTypography-body1"
                      >
                        <p
                          class="MuiTypography-root makeStyles-timeText-100 MuiTypography-body1"
                        >
                          13:45 14:00
                        </p>
                      </p>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="makeStyles-tbody-196"
                  role="rowgroup"
                />
              </table>
            </div>
          </div>
          <div
            aria-labelledby="scrollable-auto-tab-1"
            hidden=""
            id="scrollable-auto-tabpanel-1"
            role="tabpanel"
            style="overflow: auto;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CommonTT /> CommonTT should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-13"
>
  <div
    class="MuiBox-root MuiBox-root-14"
  >
    <p
      class="MuiTypography-root makeStyles-ttLabel-12 MuiTypography-body1"
    />
  </div>
  <div
    class="MuiBox-root MuiBox-root-17"
    data-testid="time-table"
  >
    <div
      class="makeStyles-root-18"
    >
      <header
        class="MuiPaper-root MuiAppBar-root makeStyles-appBar-21 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation0"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular makeStyles-regular-23 MuiToolbar-gutters makeStyles-gutters-22"
        >
          <h6
            class="MuiTypography-root makeStyles-title-20 MuiTypography-h6"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined KenButton__KenButtonDefault-sc-1sg8ful-0 makeStyles-headerBtn-16 KenButton__KenButtonSecondary-sc-1sg8ful-1 juGKMU MuiButton-outlinedPrimary"
            data-testid="button"
            label="[object Object]"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                  />
                </svg>
              </span>
              <p
                class="MuiTypography-root makeStyles-btnLabels-15 MuiTypography-body1"
              >
                labels:Share
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined KenButton__KenButtonDefault-sc-1sg8ful-0 makeStyles-headerBtn-16 KenButton__KenButtonSecondary-sc-1sg8ful-1 juGKMU MuiButton-outlinedPrimary"
            data-testid="button"
            label="[object Object]"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <img
                  src="IMAGE_MOCK"
                />
              </span>
              <p
                class="MuiTypography-root makeStyles-btnLabels-15 MuiTypography-body1"
              >
                labels:Export
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-24"
    style="display: flex; align-items: center; justify-content: flex-end;"
  >
    <div
      class="MuiBox-root MuiBox-root-25"
      style="display: flex; align-items: baseline;"
    >
      <div
        class="MuiBox-root MuiBox-root-26 makeStyles-dateSelector-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-arrow-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
      <div
        class="MuiFormControl-root-35 MuiTextField-root-34"
        format="MMMM do"
        inputsize="10"
        selector="week"
      >
        <label
          class="MuiFormLabel-root-47 MuiInputLabel-root-39 MuiInputLabel-formControl-41 MuiInputLabel-animated-44 MuiInputLabel-shrink-43 MuiFormLabel-filled-49"
          data-shrink="true"
        />
        <div
          class="MuiInputBase-root-62 MuiInput-root-51 MuiInput-underline-54 MuiInputBase-formControl-63 MuiInput-formControl-52"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-70 MuiInput-input-58"
            readonly=""
            size="10"
            type="text"
            value="AUGUST-08-2021 TO AUGUST-14-2021"
          />
        </div>
        
      </div>
      <div
        class="MuiBox-root MuiBox-root-96 makeStyles-dateSelector-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-arrow-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-97"
  >
    <p
      class="MuiTypography-root makeStyles-ttPlaceholder-11 MuiTypography-body1"
    >
      No timetable available
    </p>
  </div>
</div>
`;

exports[`<TTGrid /> TTGrid should render and match the snapshot 1`] = `
<table
  class="makeStyles-table-269"
  role="table"
>
  <thead />
  <tbody
    class="makeStyles-tbody-266"
    role="rowgroup"
  />
</table>
`;

exports[`<TTHeader /> TTHeader should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-202"
  data-testid="time-table"
>
  <div
    class="makeStyles-root-203"
  >
    <header
      class="MuiPaper-root MuiAppBar-root makeStyles-appBar-206 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-regular-208 MuiToolbar-gutters makeStyles-gutters-207"
      >
        <h6
          class="MuiTypography-root makeStyles-title-205 MuiTypography-h6"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined KenButton__KenButtonDefault-sc-1sg8ful-0 makeStyles-headerBtn-201 KenButton__KenButtonSecondary-sc-1sg8ful-1 juGKMU MuiButton-outlinedPrimary"
          data-testid="button"
          label="[object Object]"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                />
              </svg>
            </span>
            <p
              class="MuiTypography-root makeStyles-btnLabels-200 MuiTypography-body1"
            >
              labels:Share
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined KenButton__KenButtonDefault-sc-1sg8ful-0 makeStyles-headerBtn-201 KenButton__KenButtonSecondary-sc-1sg8ful-1 juGKMU MuiButton-outlinedPrimary"
          data-testid="button"
          label="[object Object]"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <img
                src="IMAGE_MOCK"
              />
            </span>
            <p
              class="MuiTypography-root makeStyles-btnLabels-200 MuiTypography-body1"
            >
              labels:Export
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</div>
`;

exports[`<WeekPicker /> WeekPicker should render and match the snapshot 1`] = `
<div
  class="MuiFormControl-root-217 MuiTextField-root-216"
  format="MMMM do"
>
  <div
    class="MuiInputBase-root-232 MuiInput-root-221 MuiInput-underline-224 MuiInputBase-formControl-233 MuiInput-formControl-222"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input-240 MuiInput-input-228"
      readonly=""
      size="40"
      type="text"
      value="August 13th"
    />
  </div>
  
</div>
`;
